adminSiteClient/SuggestedChartRevisionApproverPage.tsx

```diff
@@ -0,0 +1,1367 @@
+    }
+
+    @computed get prevBtnIsDisabled() {
+        return !this._isGraphersSet || this.rowNumValid <= 1
+    }
+
+    @computed get nextBtnIsDisabled() {
+        return !this._isGraphersSet || this.rowNumValid >= this.numTotalRows
+    }
+
+    @computed get grapherBounds() {
+        let bounds
+        if (this.previewMode === "mobile") {
+            bounds = new Bounds(0, 0, 360, 500)
+        } else {
+            if (this.desktopPreviewSize === "small") {
+                bounds = new Bounds(0, 0, 600, 450)
+            } else {
+                bounds = new Bounds(0, 0, 800, 600)
+            }
+        }
+        return bounds
+    }
+
+    @computed get rowNumValid() {
+        return Math.max(Math.min(this.rowNum, this.numTotalRows), 1)
+    }
+
+    @computed get updateButtonsIsDisabled() {
+        return !this._isGraphersSet
+    }
+
+    @computed get approveButtonIsDisabled() {
+        return (
+            this.updateButtonsIsDisabled ||
+            !this.suggestedChartRevision.canApprove
+        )
+    }
+
+    @computed get rejectButtonIsDisabled() {
+        return (
+            this.updateButtonsIsDisabled ||
+            !this.suggestedChartRevision.canReject
+        )
+    }
+
+    @computed get flagButtonIsDisabled() {
+        return (
+            this.updateButtonsIsDisabled || !this.suggestedChartRevision.canFlag
+        )
+    }
+
+    @computed get listMode() {
+        const { suggestedChartRevisionId } = this.props
+        return !suggestedChartRevisionId
+    }
+
+    @action.bound async refresh() {
+        this.clearDecisionReasonInput()
+        await this.fetchGraphers()
+        await this.fetchRefs()
+    }
+
+    @action.bound async fetchGraphers() {
+        const { admin } = this.context
+        const { suggestedChartRevisionId } = this.props
+        if (suggestedChartRevisionId === undefined) {
+            const json = await admin.getJSON("/api/suggested-chart-revisions", {
+                limit: 1,
+                offset: this.offset,
+                status:
+                    this.listMode && this.showPendingOnly
+                        ? SuggestedChartRevisionStatus.pending
+                        : null,
+                sortBy: this.sortBy,
+                sortDirection: this.sortDirection,
+            })
+            this.numTotalRows = json.numTotalRows
+            this.suggestedChartRevision = json.suggestedChartRevisions[0]
+        } else {
+            const json = await admin.getJSON(
+                `/api/suggested-chart-revisions/${suggestedChartRevisionId}`
+            )
+            this.suggestedChartRevision = json.suggestedChartRevision
+        }
+        this.decisionReasonInput = this.suggestedChartRevision.decisionReason
+            ? this.suggestedChartRevision.decisionReason
+            : ""
+        this.rerenderGraphers()
+    }
+
+    @action.bound async rerenderGraphers() {
+        this._isGraphersSet = false
+        setTimeout(() => {
```

Comment:
Why is this an imperative function that uses setTimeout? Could this not be solved with a computed property? I haven't followed the dependency chain to see if this would be difficult, just wondering about this as I read the code

Reply:
I initially tried getting this to work with a computed property but was having trouble getting the `<Grapher />` component to re-render when the `suggestedChartRevision` state changed. Not sure what the component lifecycle problem is here.

This is similar to an issue in the chart editor where clicking on the "mobile" vs. "desktop" view does not induce a re-render.